icu4j/main/classes/core/src/com/ibm/icu/impl/units/MeasureUnitImpl.java

```diff
@@ -66,11 +66,11 @@ public static MeasureUnitImpl forCurrencyCode(String currencyCode) {
         return result;
     }
 
-    public MeasureUnitImpl clone() {
+    public MeasureUnitImpl copy() {
```

Comment:
clone() to return a type other than Object is troublesome. Changing return type to Object is OK, but it's not a good convention for a class not implementing Cloneable. I think changing method name to something else would be the easiest change without introducing any confusions. Anyway, this class is in impl package, so we don't really need to worry about it.

Reply:
Hm, I thought we had long ago adopted covariant return types in ICU4J. I just checked a couple of classes (UnicodeSet, DecimalFormat) and am surprised to see that their clone() methods return Object. Why is that?

(E.g. https://blogs.oracle.com/sundararajan/covariant-return-types-in-java "Object.clone method returns Object type. A subclass such as Shape's clone method can return Shape type instead of Object type.")

Not arguing against this change (especially if these classes are not Cloneable), but I would like to learn.

Reply:
BytesTrie.clone() returns BytesTrie... https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/util/BytesTrie.html